### PR TITLE
Update release repositories for ROS 2 releases from tier4.

### PR DESCRIPTION
**Pull request by nuclearsandwich**
Responding to the request in https://github.com/ros/rosdistro/pull/31289#issuecomment-1006495518
This updates the release repository url for ROS 2 distros to all use the
ros2-gbp copy of the repository.

Since the same osqp_vendor-release repository is used for ROS Noetic as
well we could either update that url and use the ros2-gbp repository for
all distributions or maintain the existing release repository just for
ROS Noetic.

### CODE REVIEW

**Comment by nuclearsandwich** [comment]
@wep21 can you please confirm that you have access to the two new release repositories and approve these changes.

Fellow rosdistro maintainers, this change shouldn't be merged without approval from @wep21 or someone else from the tier4 organization on Github.

**Comment by wep21** [comment]
@nuclearsandwich Thank you for creating the repositories. I can access both repositories, so I approve these changes with this comment. Also, I appreciate it if you could invite my colleague @kenji-miyake to ros2-gbp/tier4 team.

**Comment by nuclearsandwich** [comment]
> Also, I appreciate it if you could invite my colleague @kenji-miyake to ros2-gbp/tier4 team.

I've added them as well. Their invitation should be deployed shortly.

**Review by clalancette — APPROVED** [review]

**Pull request merged**
